docs/cookbook/syncing/full-scan-vs-sync.md

```diff
@@ -0,0 +1,17 @@
+# Full Scan vs Sync
+
+The BDK libraries rely on two syncing concepts we call _**Full Scan**_ and _**Sync**_. This page explains the difference between the two and recommendations on when to use each.
+
+## Full Scan
+
+A _full scan_ is an operation that aims to build a complete picture of a wallet's UTXOs from scratch, with no prior data or knowledge about the wallet required. It involves querying the blockchain data client for ScriptPubKeys that the wallet owns, typically at a number of indices on two keychains, an external keychain and an internal one (often called a _change_ keychain). Because the wallet doesn't know which scripts/addresses have been given out and potentially have funds associated with them, it needs to query the blockchain client for a group of addresses, see if any of those have balances, and keep asking for more until a number of addresses in a row for each keychain are unused (the exact number is configurable and is what's typically known as a _stop gap_). The operation also returns an new chain tip the wallet uses to update its [`LocalChain`](https://docs.rs/bdk_chain/latest/bdk_chain/local_chain/struct.LocalChain.html).
```

Comment:
I would move `stop_gap` out to its own H2 since it will show up in "sync" as well if you don't do the "full scan" before calling "sync".
I would also highlight some common stop gap values that some wallets use: Electrum, Sparrow etc. I would even suggest a number like 100 for personal use and 1,000 for more crazy stuff as a good rule of thumb.